examples/barotropic_instability.jl

```diff
@@ -0,0 +1,288 @@
+
+# # Eigenplotting
+#
+# A good algorithm wouldn't be complete without a good visualization tool,
+
+using Oceananigans.AbstractOperations
+
+u, v, w = model.velocities
+
+vorticity = ComputedField(∂x(v) - ∂y(u))
+
+x, y, z = nodes(vorticity)
+
+eigentitle(σ, t) = "Iteration $(length(σ)): most unstable eigenmode"
+eigentitle(::Nothing, t) = @sprintf("Vorticity at t = %.2f", t)
+
+function eigenplot!(ω, σ, t; ω_lim=maximum(abs, ω)+1e-16)
+    
+    background_flow = plot(bickley_jet.(0, y, 0, 0), y, label=nothing,
+                           ylims=(grid.yF[1], grid.yF[grid.Ny]), xlabel="U(y)", ylabel="y")
+
+    ω_contours = contourf(x, y, clamp.(ω, -ω_lim, ω_lim)';
+                          color = :balance, aspectratio = 1,
+                          levels = range(-ω_lim, stop=ω_lim, length=21),
+                          xlims = (grid.xF[1], grid.xF[grid.Nx]),
+                          ylims = (grid.yF[1], grid.yF[grid.Ny]),
+                          clims = (-ω_lim, ω_lim), linewidth = 0)
+                          
+    eigenplot = plot(background_flow, ω_contours,
+                     layout = Plots.grid(1, 2, widths=[0.2, 0.8]),
+                     link = :y, size = (600, 200),
+                     title = ["Background flow" eigentitle(σ, t)])
+
+    return eigenplot
+end
+
+# # Rev your engines...
+#
+# We initialize the power iteration with random noise.
+# The amplitude of the initial condition is arbitrary since our algorithm
+# will rescale the velocity field iteratively until the simulation's stop_criteria
+# is no longer met.
+
+using Random, Statistics, Oceananigans.AbstractOperations
+
+mean_perturbation_energy = mean(1/2 * (u^2 + v^2), dims=(1, 2, 3))
+
+noise(x, y, z) = randn()
+
+set!(model, u=noise, v=noise)
+
+growth_rates = estimate_growth_rate!(simulation, mean_perturbation_energy, vorticity)
+
+@info "\n Power iterations converged! Estimated growth rate: $(growth_rates[end]) \n"
+
+# # Plot the result
+
+scatter(filter(σ -> isfinite(σ) && σ > 0, growth_rates),
+        xlabel = "Power iteration",
+        ylabel = "Growth rate (1/s)",
+        yscale = :log10,
+         label = nothing)
+
+# # The fun part
+#
+# Now for the fun part: simulating the transition to turbulence.
+
+estimated_growth_rate = growth_rates[end]
+
+using Oceananigans.OutputWriters
+
+simulation.output_writers[:vorticity] = JLD2OutputWriter(model, (ω=vorticity,),
+                                                         schedule = IterationInterval(20),
+                                                         prefix = "barotropic_instability",
+                                                         force = true)
+
+pop!(simulation.stop_criteria) # Remove the vorticity_exceeds_threshold stop_criteria
+
+model.clock.iteration = 0
+model.clock.time = 0
+simulation.stop_iteration = 2000 #stop_time = 10 / growth_rates[end]
+
+@info "*** Running a simulation of barotropic instability..."
+
+rescale!(model.velocities, factor=10)
+run!(simulation)
```

Comment:
Just to make sure: these lines now simply rescale the latest `model.velocities` (which are of the form of the eigenmode) and let the simulation run without stopping it this time, correct?

Reply:
Yes, `rescale!` is also used above in the power method algorithm.

Reply:
So it's the same `model` that was used for the power method that you let go to see the nonlinear equilibration after we change the `stop_criteria` above. (That was mainly my question.)

Reply:
Indeed!

I think your suggestions was acutually to change the problem (removing the `BackgroundFlow`). This is doable for sure, but makes the example a bit more complicated, so I opted for something simple here (simulating just the nonlinear evolution of the perturbation field).

Reply:
No, my suggestion was to do precisely what you've done here!
I would only also plot the time-series of energy together with exp(2σt). I'll add this...